File: intro/week6/psets6/mario.py
# write a program using python to print a pyramid similar to that of mario with #s

#   Height: 4
#      #  #
#     ##  ##
#    ###  ###
#   ####  ####

from cs50 import get_int

def main():
    #take input for height
    while True:
        height = get_int("Height: ")
        if height > 0 and height <= 8:
            break

    for storey in range(1, height + 1):     # storey stores the current line in pyramid with top being 1 and last being height
        for _ in range(height - storey):    # this will put spaces as required before starting to print #s
            print(" ", end = "")
        for k in range(2):      # to repeat printing #s with a space in between
            for _ in range(storey):     # will print spaces required number of times
                print("#", end="")
            if k < 1:
                print(" ", end="")
        print()        

if __name__ == "__main__":
    main()

File: intro/week1/mario.c
// program to print a pyramid for mario (Pset1)
// Yuvraj S Jadon
#include </media/yuvrajsj18/New Volume/cs50/cs50.c>
#include <stdio.h>

int main(void)
{
    int height;
    // Take input as long as it is not in range 1-8
    do
    {
        height = get_int("Height: ");
    }
    while(height < 1 || height > 8);
       
    for (int i = 1; i <= height; i++)            // i measures height in each run of this loop
    {
        for(int s = 0; s < height - i; s++)    // s represents the number of spaces before first '#'
             printf(" ");
        for(int k = 0; k < 2; k++)                // this loop with k gives the shape of pyramid
        {
            for(int n = 0; n < i; n++)          // n represents the number of '#' each line will have
                printf("#");
            if(k < 1)
                printf("  ");
        }
        printf("\n");    
    }
}
